app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,48 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+  def Show
+
+  end
+  def new
+    @post = Post.find_by_id(params[:post_id])
+  end
+
+  def create
+
+    p "here is there"
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.create(comment_params) do |comment|
+      comment.user_id=current_user[:id]
```

Comment:
These line could break. Read the API doc for `find` method. This method will throw exception. Are you catching any?
And why are you using loop. Find some other way to accomplish this task.

Reply:
I am not tying to do catching.
Done.

Reply:
Rails has its own mechanism for caching the query result. Since you are creating record. How will the cache result will help you. Creating something is destructive and transaction operation. Using cache in such case is not a good idea.